File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let data = s.split('')
    let stack = []
    
    for(let item of data){
        if(stack.length === 0 || item === '('){
            stack.push(item)
        }
    
        else if (item === ")"){
            stack.pop()
        }
        
    }

    
    
    
    return stack.length === 0;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = []
    idx =0
    for i in range(len(s)) :
        if(i==0 and s[i] ==")"):
            return False
        
        
        if s[i] == "(":
            answer.append('(')
        elif s[i] ==")" and len(answer)>0:
            answer.pop()
        
    if(len(answer) == 0):
        return True
    
    return False